lib/issuer.js

```diff
@@ -188,33 +230,206 @@ api.claim = function(actor, id, callback) {
+      return next(new BedrockError('Invalid JSON.', 'JSONParseError', {
+        httpStatusCode: 400,
+        public: true
+      }));
     }
-  }, function(err) {
-    callback(err);
+    return next();
+  }
+
+  // FIXME: determine proper endpoint
+  app.get('/creds/claimid/:claimid', function(req, res, next) {
+    var claimId = req.params.claimid;
+    findCredentialByClaimId(claimId, function(err, results) {
+      if(err) {
+        return next(new BedrockError(
+          'The credential could not be located.',
+          'CredentialLookupFailed', {
+            httpStatusCode: 400,
+            'public': true
+          }, err));
+      }
+      return res.json(results);
+    });
+  });
+  // FIXME: this is a provisional endpoint and authentication is required
+  app.put('/creds/:id', function(req, res, next) {
+    var credentialId = req.params.id;
+    console.log('BODY *****', credentialId, req.body);
+    var now = Date.now();
+    if(req.body.accepted) {
+      var updateQuery = {id: credentialId};
+      var updateParams = {
+        $set: {
+          'meta.updated': now,
+          'meta.acceptance': {
+            status: 'accepted',
+            date: now
+          }
+        }
+      };
+      store.provider.collection.update(
+        updateQuery, updateParams, function(err, result) {
+          if(err) {
+            return next(new BedrockError(
+              'The credential could not be updated.',
+              'UpdateCredentialFailed', {
+                httpStatusCode: 400,
+                'public': true
+              }, err));
+          }
+          return res.json({status: 'OK'});
+        });
+    } else {
+      return res.json({status: 'Nothing Updated'});
+    }
+  });
+  // FIXME: This is a provisional endpoint
+  app.post(
+    '/credential-login', parseForm, parseJsonData, function(req, res, next) {
+      console.log('BODY:', req.body);
+      var identity = req.body;
+      views.getDefaultViewVars(req, function(err, vars) {
+        // FIXME: this is likely the identity returned by the authentication
+        // or at least verified signatures etc.
+        vars.issuer = {};
+        vars.issuer.identity = identity;
+        vars.issuer.unsignedCredentialEndpoint =
+          config.issuer.endpoints.unsignedCredential;
+        res.render('index.html', vars);
+      });
   });
-});
 
-bedrock.events.on('bedrock.test.configure', function() {
-  // load test config
-  require('./test.config');
-});
+  app.post(config.issuer.endpoints.unsignedCredential,
+    [checkHeaders, validate({body:'credential'}), ensureAuthenticated],
+    function(req, res, next) {
+      // FIXME: Replace credential signing key from config with key from db
+      var options = {};
+      options.privateKey = config.issuer.credentialSigningPrivateKey;
+      options.publicKeyId = config.issuer.credentialSigningPublicKeyId;
+      var credential = req.body;
+      api.issue(null, credential, options, function(err) {
```

Comment:
Add a FIXME that indicates `null` should be filled out with the `actor` from the authenticated request. Right now, permission checks are disabled, but we need to do a sweep and enable them later.